src/basic/iovec-util.c

```diff
@@ -63,6 +63,8 @@ char* set_iovec_string_field_free(struct iovec *iovec, size_t *n_iovec, const ch
 }
 
 void iovec_array_free(struct iovec *iovec, size_t n) {
+        assert(iovec || n == 0);
```

Comment:
Note, `FOREACH_ARRAY()` handles the condition that trigger this assertion gracefully, IIRC.
Though, this assertion itself is good.